### ISSUE TITLE
Generate a config yaml schema to better support config building/editing

### ISSUE THREAD

**Issue by anussel5559** [LucasPickering/slumber#374]
**Did you [search](https://github.com/LucasPickering/slumber/issues) for existing issues already?**
Yup!

**Is your feature request related to a problem? Please describe.**
Nope

**Describe the solution you'd like**
Currently, when building or editing a slumber yaml configuration, there's no schema available - this isn't a huge deal, but leaves the user to their own devices as far as building a valid config file.

With a generated schema, the user's IDE can help guide them through building a valid config, and call out potential issues before they go to run the app.

**Describe alternatives you've considered**
None

**Additional context**
I looked a bit in to [schemars](https://github.com/GREsau/schemars) to generate the schema from the `Collection` struct, but there was enough sprawl there that it made more sense to bring it up and talk about this potential feature more generally.

**Comment by anussel5559** [comment]
Sure! So my thought is it'd be nice to _not_ have to copy-paste. The IDE can be powerful if it knows what the schema is, it's mostly useless without it.

Today, unless you copy-paste and don't make any mistakes when modifying that pasted bit of config, you'd never know that you may have an invalid config file until you startup the app when it parses. This feels slightly wasteful since the IDE could've told the user that well before running the app.

Additionally, these configs make use of custom yaml tags, which the generic YAML validator knows nothing about (and thus complains) - 
![image](https://github.com/user-attachments/assets/9082bcba-6e6f-4205-8ab1-662510d8422f)

Ultimately this is a minor inconvenience and FAR from a boundary to successfully writing a valid config.

**Comment by anussel5559** [comment]
@LucasPickering yup, yaml schema is just JSON schema and that's what schemars would generate.

There's a few ways the JSON schema can be associated to the yaml file to enable validation/hinting/docs/etc... (checkout [this article](https://developers.redhat.com/blog/2020/11/25/how-to-configure-yaml-schema-to-make-editing-files-easier#yaml_schema)), but likely the easiest way is to host the schema, and document that schema validation can be enabled by setting:
```yaml
# yaml-language-server: $schema=https://my.url.to/the/schema
```
at the top of the config file.

**Comment by LucasPickering** [comment]
Yeah I like this idea! I just opened a related feature #376 to add a subcommand `slumber new`, which would generate a new collection file. This file would include that comment at the top so users don't have to write it manually. And we could host the YAML file on the website, since that is already a process that generates static files in CI and hosts them.

**Comment by anussel5559** [comment]
awesome! That makes a lot of sense. I'm definitely interested in helping out with this, likely after I get some solid movement on the collection ref stuff.

**Comment by LucasPickering** [comment]
I didn't forget about this! Just put up [a PR](https://github.com/LucasPickering/slumber/pull/533) to add this. It's part of a major collection format overhaul in 4.0, so it'll still be a bit before it gets released.
